renovate/generic-regex-managers.json5

```diff
@@ -0,0 +1,36 @@
+{
+    "$schema": "https://docs.renovatebot.com/renovate-schema.json",
+    "customManagers": [
+        {
+            // Searches through Containerfiles to find ARG and ENV variables associated with git references
+            // and updates them to the latest commit digest.
+            // The regex captures the dependency name, package name, git reference, versioning strategy,
+            // and current digest from the renovate directive.
+            // ref: https://docs.renovatebot.com/modules/manager/regex/#advanced-capture
+            "customType": "regex",
+            "fileMatch": [
+                "Containerfile$"
+            ],
+            "matchStrings": [
+                "#\\s*renovate:\\s*datasource=git-refs\\sdepName=(?<depName>\\S+)\\s(?:packageName=(?<packageName>\\S+)\\s)?(?:gitRef=(?<currentValue>\\S+)\\s)?(?:versioning=(?<versioning>\\S+)\\s)?(?:extractVersion=\\S+\\s)?type=(?<type>digest)\\s(?:ARG|ENV)\\s.*?=(?<currentDigest>\\S+)"
+            ],
+            "datasourceTemplate": "git-refs",
+            "versioningTemplate": "{{#if versioning}}{{{versioning}}}{{else}}semver{{/if}}"
+        },
+        {
+            // Searches through Containerfiles to find ARG and ENV variables associated with various datasources
+            // and updates them to the latest version.
+            // The regex captures the dependency name, package name, versioning strategy,
+            // and current version value from the renovate directive.
+            // ref: https://docs.renovatebot.com/modules/manager/regex/#advanced-capture
+            "customType": "regex",
+            "fileMatch": [
+                "Containerfile$"
+            ],
+            "matchStrings": [
+                "#\\s*renovate:\\s*datasource=(?<datasource>.*?)\\sdepName=(?<depName>\\S+)\\s(?:packageName=(?<packageName>\\S+)\\s)?(?:versioning=(?<versioning>\\S+)\\s)?(?:extractVersion=\\S+\\s)?(?:ARG|ENV)\\s.*?=(?<currentValue>\\S+)"
+            ],
+            "versioningTemplate": "{{#if versioning}}{{{versioning}}}{{else}}semver{{/if}}"
+        }
+    ]
```

Comment:
These are two separate managers since the git-refs datasource requires the current version to be specified under the "currentDigest" regex group, which is the opposite to almost all other datasources.

Reply:
I like the motivation here, however I'm uncertain how applicable these presets are for Podman/Buildah/Skopeo (and dependencies) - the primary users of this file.  I also a bit nervous because in my mind, those repos may have quite a few `Containerfile` in use (for example) for testing purposes which should never ever be updated (since they're likely tied to the test-environment context).  So some research is needed to determine how many downstream users of these presets will be affected.  Also how many exclusions will be needed and in which repos.

Otherwise, I appreciate the helpful comments, and docs references.  Would you consider  de-duplicating the second instance?  You can just add something like "see custom Containerfile regex manager config comments above".

Reply:
Ahh yes, you're right, that's good.  So little/no risk of causing problems w/ existing users then.

I originally only intended these default presets to be used for Podman/Buildah/Skopeo (and dependencies).  It's fine if they're used elsewhere, but there may be future expectation mismatches.  Esp. WRT ongoing/future maintenance in those repos, but I'm leaving that up to Jan/Lokesh since they're looking after this stuff going forward.

There are some `Containerfile` in `containers/automation_images` that could make use of these additions, but this too is now out of my hands.  In any case, I'm not going to block this PR, and I don't see it as risky as I initially thought.